### PR TITLE
Simpler port

### PR DESCRIPTION
**Pull request by Wosin**
An idea to modify the current Config and Server layers to make it easier to start server on custom port.

### CODE REVIEW

**Review by adamgfraser — CHANGES_REQUESTED** [review]
The `()` after every usage of `Server.default` is very unidiomatic since this would normally indicate that this is a side effecting method. It also means that the `default` server is not a value that will automatically be memoized and shared during layer construction.

This can already be done as:

```scala
ServerConfig.live(ServerConfig.default.port(???)) >>> Server.live
```

This can potentially be shortened by introducing an overloaded `live` constructor on `Server` that takes a `ServerConfig` so the user does not need to manually compose the layers:

```scala
def live(serverConfig: ServerConfig): ZLayer[Any, Throwable, Server] =
  ServerConfig.live(serverConfig) >>> Server.live

Server.live(ServerConfig.default.port(???))
```

Or going a step further:

```scala
def defaultWith(f: ServerConfig => ServerConfig): ZLayer[Any, Throwable, Server] =
  live(f(ServerConfig.default))

Server.defaultWith(_.port(???))
```

Or even:

```scala
def defautlWithPort(port: Int): ZLayer[Any, Throwable, Server] =
  defaultWith(_.port(port))

Server.defaultWithPort(???)
```

These would not impact the ergonomics of the existing usage of the server and would support other modifications to the server configuration in a compositional way in addition to just the port.

**Comment by Wosin** [comment]
Okay, that definitely looks better. Let me fix that.

File: zio-http/src/main/scala/zio/http/Server.scala [hunk 1]
```diff
@@ -6,7 +6,7 @@ import zio.http.netty.server._
 import zio.stacktracer.TracingImplicits.disableAutoTrace // scalafix:ok;
 
 trait Server {
-  def install[R](httpApp: App[R], errorCallback: Option[ErrorCallback] = None)(implicit
+  def   install[R](httpApp: App[R], errorCallback: Option[ErrorCallback] = None)(implicit
```
Comment (adamgfraser) [inline zio-http/src/main/scala/zio/http/Server.scala]: Stray formatting change?

File: zio-http/src/main/scala/zio/http/Server.scala [hunk 2]
```diff
@@ -30,11 +30,21 @@ object Server {
...
+
+  def defaultWith(f: ServerConfig => ServerConfig): ZLayer[Any, Throwable, Server] =
+    live(f(ServerConfig.default))
+
   val default: ZLayer[Any, Throwable, Server] = {
     implicit val trace = Trace.empty
     ServerConfig.live >>> live
   }
 
+  def live(conf: ServerConfig) = {
```
Comment (adamgfraser) [inline zio-http/src/main/scala/zio/http/Server.scala]: Very minor but no need for curly braces here.

File: zio-http/src/test/scala/zio/http/api/ServerClientIntegrationSpec.scala [hunk 3]
```diff
@@ -55,7 +55,7 @@ object ServerClientIntegrationSpec extends ZIOSpecDefault {
       },
     ).provide(
       Server.live,
-      ServerConfig.live,
+      ServerConfig.live(),
```
Comment (adamgfraser) [inline zio-http/src/test/scala/zio/http/api/ServerClientIntegrationSpec.scala]: Don't think this change should be necessary.

File: zio-http/src/main/scala/zio/http/ServerConfig.scala [hunk 4]
```diff
@@ -122,9 +122,9 @@ final case class ServerConfig(
 }
 
 object ServerConfig {
-  val default: ServerConfig = ServerConfig()
+  def default: ServerConfig = ServerConfig()
```
Comment (adamgfraser) [inline zio-http/src/main/scala/zio/http/ServerConfig.scala]: Don't think these changes should be necessary.

File: zio-http/src/main/scala/zio/http/Server.scala [hunk 5]
```diff
@@ -30,11 +30,22 @@ object Server {
...
+
+  def defaultWith(f: ServerConfig => ServerConfig): ZLayer[Any, Throwable, Server] =
+    live(f(ServerConfig.default))
+
   val default: ZLayer[Any, Throwable, Server] = {
     implicit val trace = Trace.empty
     ServerConfig.live >>> live
   }
 
+  def live(conf: ServerConfig): ZLayer[Any, Throwable, Server] = {
```
Comment (adamgfraser) [inline zio-http/src/main/scala/zio/http/Server.scala]: If this is already going to be a `def` we can just accept a `Trace` as a parameter.

**Review by adamgfraser — APPROVED** [review]
Looks good!

File: zio-http/src/main/scala/zio/http/Server.scala [hunk 6]
```diff
@@ -30,11 +30,21 @@ object Server {
...
+
+  def defaultWith(f: ServerConfig => ServerConfig)(implicit trace: Trace): ZLayer[Any, Throwable, Server] =
+    live(f(ServerConfig.default))
+
   val default: ZLayer[Any, Throwable, Server] = {
     implicit val trace = Trace.empty
     ServerConfig.live >>> live
   }
 
+  def live(conf: ServerConfig)(implicit trace: Trace): ZLayer[Any, Throwable, Server] = {
```
Comment (adamgfraser) [inline zio-http/src/main/scala/zio/http/Server.scala:44]: Don't need the curly braces here.

**Review by jdegoes — APPROVED** [review]

**Pull request merged**